highlevel/robot/gateway/motor.py

```diff
@@ -1,73 +1,156 @@
 """
 Motor gateway module.
 """
-from dataclasses import dataclass
 
 from highlevel.adapter.socket import SocketAdapter
 from highlevel.logger import LOGGER
-from highlevel.robot.entity.type import TickPerSec
+from highlevel.robot.entity.configuration import Configuration
+from highlevel.robot.entity.type import TickPerSec, Tick
+from highlevel.util.filter.pid import PIDConstants
 from proto.gen.python.outech_pb2 import BusMessage, MoveWheelAtSpeedMsg, PIDCoefficients, \
-    PIDConfigMsg
-
-
-@dataclass(frozen=True)
-class PIDValues:
-    """
-    Structure holding a PID's constants
-    """
-    k_p: float
-    k_i: float
-    k_d: float
-
-
-FIXED_POINT_COEF = 2**16
+    PIDConfigMsg, WheelPositionTargetMsg, WheelControlModeMsg, WheelTolerancesMsg, WheelPWMMsg
 
 
 class MotorGateway:
     """
     Motor gateway.
     """
-    def __init__(self, motor_board_adapter: SocketAdapter):
+    def __init__(self, motor_board_adapter: SocketAdapter,
+                 configuration: Configuration):
+        self.configuration = configuration
         self.motor_board_adapter = motor_board_adapter
+        self.pid_speed_left = PIDConstants(0, 0, 0)
```

Comment:
This gateway seems stateful, why is that?

Usually gateways are just... gateways to the hardware you should not expect it to hold any state, it provides functions to communicate with a lower level and that's it